### PR TITLE
Download launcher from release instead of rebuilding

### LINKED ISSUE #497: Pull pre-compiled NorthstarLauncher instead of building from source on release
As mentioned in https://github.com/R2Northstar/NorthstarLauncher/pull/467#issuecomment-1588134194 with the merge of https://github.com/R2Northstar/NorthstarLauncher/pull/467 we need to also update release workflow as the old version would just pull in the launcher source code and try to compile it using the solution file. Naturally with the change to cmake this also needs to be update.

Instead of building from source and having to update again the next time the build system changes again, we should instead just pull in the precompiled files that are pushed on each release via https://github.com/R2Northstar/NorthstarLauncher/pull/439

### PR DESCRIPTION
**Pull request by F1F7Y**
Fetches launcher build from latest release on NorthstarLauncher

NOTE: Due to the differences between old build system and cmake for the files to be structured properly a release on NorthstarLauncher needs to be made.

Closes #497

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,53 +17,15 @@ env:
...
     steps:
       - name: Download compiled launcher
-        uses: actions/download-artifact@v3
+        uses: dsaltares/fetch-gh-release-asset@master
         with:
-          name: northstar-launcher
-          path: northstar-launcher
+          repo: 'R2Northstar/NorthstarLauncher'
+          file: northstar-launcher.zip
+          token: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (F1F7Y) [reply .github/workflows/build.yml:28]: `wget` was the first thing i wanted to do, but couldnt figure out how to easily get latest release, will look into it

**Review by GeckoEidechse — APPROVED** [review]
Builds fine based on https://github.com/F1F7Y/Northstar/actions/runs/5349845687 and folder structure of resulting zip also matches `v1.14.3`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -98,15 +57,13 @@ jobs:
...
-          unzip northstar-discord-rpc.zip
-          mv -v DiscordRPC.dll northstar/R2Northstar/plugins
-
-          mv -v northstar-launcher/wsock32.dll northstar/bin/x64_retail
+          unzip northstar-discord-rpc.zip -d northstar/R2Northstar/plugins
+          
           unzip NorthstarStubs.zip -d northstar/bin/x64_dedi
-
-          mv -v northstar-launcher/* northstar
+
```
Comment (GeckoEidechse) [inline .github/workflows/build.yml:65]: Trailing whitespace >:(
```suggestion

```

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pg9182) [reply .github/workflows/build.yml]: Use the official actions-script action (see northstar-dedicated-ci for some examples).

**Pull request merged**